File: techconnect/lib/models/technician.dart
class Technician {
  final int id;
  final String name;
  final String email;
  final String phone;
  final String domain;
  final String city;
  final String? description;
  final String? profileImageUrl;
  final double rating;
  final int reviewCount;
  final DateTime createdAt;

  Technician({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    required this.domain,
    required this.city,
    this.description,
    this.profileImageUrl,
    required this.rating,
    required this.reviewCount,
    required this.createdAt,
  });

  factory Technician.fromJson(Map<String, dynamic> json) {
    return Technician(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      phone: json['phone'],
      domain: json['domain'],
      city: json['city'],
      description: json['description'],
      profileImageUrl: json['profileImageUrl'],
      rating: (json['rating'] as num).toDouble(),
      reviewCount: json['reviewCount'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'phone': phone,
      'domain': domain,
      'city': city,
      'description': description,
      'profileImageUrl': profileImageUrl,
      'rating': rating,
      'reviewCount': reviewCount,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: techconnect_b/src/main/java/com/example/techconnect/backend/entity/Technician.java
package com.techconnect.backend.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "technicians")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Technician {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String name;

  @Column(nullable = false, unique = true)
  private String email;

  @Column(nullable = false)
  private String password;

  @Column(nullable = false)
  private String phone;

  @Column(nullable = false)
  private String domain; // Electricity, Plumbing, IT, etc.

  @Column(nullable = false)
  private String city;

  @Column(length = 1000)
  private String description;

  private String profileImageUrl;

  @Column(nullable = false)
  private Double rating = 0.0;

  @Column(nullable = false)
  private Integer reviewCount = 0;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private User.Role role = User.Role.TECHNICIAN;

  @CreationTimestamp
  private LocalDateTime createdAt;
}
